Qt 6 Building Porting

Chiitoo opened the pull request:
Builds and runs (no crashes so far!) with spell checking disabled, on Gentoo Linux with Qt 6.7.2.

Bunches to do still:

- Clean up CMakeLists once decided if Qt 5 will be supported via the same codebase.
- Some bits have been commented out, requiring more figuring out.

Emdek commented:
@Chiitoo, these changes for the QtWebEngine seem to be the biggest, but not as big as expected.
The most annoying thing seem to be that codec change for the streams.

Chiitoo commented:
Did a quick'ish rebase so that this builds again with Qt 6 (with no spellcheck), and some additional tweaks such as 'ifdef' additions but... mistakingly aborted the rebase with them so those did not get in this time around.

Chiitoo committed:
qt6: ‘const class QUrl’ has no member named ‘topLevelDomain’

- https://bugreports.qt.io/browse/QTBUG-80308

Chiitoo committed:
qt6: include 'QtWebEngineCore/QWebEngineHttpRequest'

Chiitoo committed:
qt6: no member named 'print' in 'Otter::QtWebEnginePage'

Chiitoo committed:
qt6: 's/QWebEngineSettings::globalSettings/QWebEngineSettings::defaultSettings/'

- https://code.qt.io/cgit/qt/qtwebengine.git/commit/?h=6.7&id=8daefcfd8e1bf41ae4d06c7a79a526b81a78dc58

Chiitoo committed:
qt6: move 'QWebEngineCertificateError' to 'QtWebEngineCore'

Chiitoo committed:
qt6: use 'defaultProfile->settings' instead of 'defaultSettings'

- https://code.qt.io/cgit/qt/qtwebengine.git/commit/?h=6.7&id=0870cc93ad3240f5898f231c8f8b8fd82035206d

Chiitoo committed:
qt6: move 'QWebEngineFullScreenRequest' to 'QtWebEngineCore'

Chiitoo committed:
qt6: use 'QWebEngineDownloadRequest' instead of 'QWebEngineDownloadItem'

Chiitoo committed:
qt6: move 'QtWebEngineWidgets' to 'QtWebEngineCore'

Chiitoo committed:
qt6: 's/QWebEngineDownloadItem/QWebEngineDownloadRequest/'

Chiitoo committed:
qt6: move 'QAction' to 'QtGui'

Chiitoo committed:
qt6: rename downloadrequest members

haydara commented:
I have just checked out your branch in a clean branch from otter master, I had to make some small changes and it is compiling and working, you can see my changes at:
https://github.com/haydara/otter-browser/commit/3cc281e0cb41e9cd96a51f6900fc6c8ac3598ab5

I hope it is helpful

Chiitoo commented:
Yes, indeed, the spellcheck is indeed broken still.

Never got to looking into how to best handle it (like, unsure if we even want to bundle Sonnet, though at least there should be an option to use the system-installed version instead etc.).

haydara commented:
I have also pushed the following commit to my branch https://github.com/OtterBrowser/otter-browser/commit/1c38c3e216d039d62376688504670cc9091f9553 

Fix Qt6 WebEngine session restore and start page handling
Keep the WebEngine widget visible during the start page transition to avoid
the first URL load flicker on Qt6.

Show the start page as an overlay instead of hiding or reparenting the web
widget. This avoids Qt6 window refresh issues and keeps Qt5 from treating the
start page as a splitter pane.

Also store Otter's private tab state explicitly instead of relying on
QWebEngineProfile::isOffTheRecord(), because Qt6 reported normal tabs as
private. This caused normal tabs to be skipped during session saving, resulting
in an empty saved session.

haydara commented:
I think with these commits is your Qt6 port ready to merge into the otter repository. I think we should have Qt6 support as soon as possible because the webengine shipped with qt5 very outdated and modern websites does not look correctly with it
